src/windows/disk.rs

```diff
@@ -344,3 +382,47 @@ fn os_string_from_zero_terminated(name: &[u16]) -> OsString {
     let len = name.iter().position(|&x| x == 0).unwrap_or(name.len());
     OsString::from_wide(&name[..len])
 }
+
+/// Returns a tuple consisting of the total number of bytes read and written by the volume with the specified device path
+fn get_disk_io(device_path: &[u16], handle: Option<HandleWrapper>) -> Option<(u64, u64)> {
+    let handle =
+        handle.or(unsafe { HandleWrapper::new_from_file(device_path, Default::default()) })?;
+
+    if handle.is_invalid() {
+        sysinfo_debug!(
+            "Expected handle to '{:?}' to be valid",
+            String::from_utf16_lossy(device_path)
+        );
+        return None;
+    }
+
+    let mut disk_perf = DISK_PERFORMANCE::default();
+    let mut bytes_returned = 0;
+
+    // SAFETY: the handle is checked for validity above
+    unsafe {
+        // See <https://learn.microsoft.com/en-us/windows/win32/api/winioctl/ni-winioctl-ioctl_disk_performance> for reference
+        DeviceIoControl(
+            handle.0,
+            IOCTL_DISK_PERFORMANCE,
+            None, // Must be None as per docs
+            0,
+            Some(&mut disk_perf as *mut _ as _),
+            size_of::<DISK_PERFORMANCE>() as u32,
+            Some(&mut bytes_returned),
+            None,
+        )
+    }
+    .map_err(|err| {
+        sysinfo_debug!("Error: DeviceIoControl(IOCTL_DISK_PERFORMANCE) = {:?}", err);
+        err
+    })
+    .ok()
+    .and_then(|_| {
+        disk_perf
+            .BytesRead
+            .try_into()
+            .ok()
+            .zip(disk_perf.BytesWritten.try_into().ok())
+    })
```

Comment:
Seems a bit more complicated than needed. What do you think about:

```suggestion
     unsafe {
        // See <https://learn.microsoft.com/en-us/windows/win32/api/winioctl/ni-winioctl-ioctl_disk_performance> for reference
        DeviceIoControl(
            handle.0,
            IOCTL_DISK_PERFORMANCE,
            None, // Must be None as per docs
            0,
            Some(&mut disk_perf as *mut _ as _),
            size_of::<DISK_PERFORMANCE>() as u32,
            Some(&mut bytes_returned),
            None,
        )
    }
    .map_err(|err| {
        sysinfo_debug!("Error: DeviceIoControl(IOCTL_DISK_PERFORMANCE) = {:?}", err);
        err
    })
    .ok()?;
    Some((disk_perf.BytesRead.try_into().ok()?, disk_perf.BytesWritten.try_into().ok()?)) 
```